### PR TITLE
ensure that we update bestie container and not the init container

### LINKED ISSUE #91: get the right container from the deployment when updating the bestie image
there are 2 pods in the deployment, make sure get the correct one.

### PR DESCRIPTION
**Pull request by skattoju**
fixes #91

### CODE REVIEW

**Review by rocrisp — APPROVED** [review]

File: internal/sub_reconcilers/reconcile_deployment_updates.go [hunk 1]
```diff
@@ -115,21 +115,25 @@ func (r *DeploymentImageReconciler) upgradeOperand(ctx context.Context, bestie *
...
+	for pos, container := range dp.Spec.Template.Spec.Containers {
+		if container.Name == "bestie" {
+			imageInDeployment = container.Image
+			positionInContainerArray = pos
 		}
-		// Level 4 Add metrics
-		bestie_metrics.ApplicationUpgradeCounter.Inc()
 	}
+	if imageInDeployment != getBestieContainerImage(bestie) {
+		r.Log.Info("Upgrade Operand")
```
Comment (yuriolisa) [inline internal/sub_reconcilers/reconcile_deployment_updates.go:127]: Could you please specify which operand is getting upgraded?

File: internal/sub_reconcilers/reconcile_deployment_updates.go [hunk 2]
```diff
@@ -115,21 +115,25 @@ func (r *DeploymentImageReconciler) upgradeOperand(ctx context.Context, bestie *
...
+		dp.Spec.Template.Spec.Containers[positionInContainerArray].Image = getBestieContainerImage(bestie)
+	}
+	err = r.client.Update(ctx, dp)
+	if err != nil {
+		r.Log.Error(err, "Deployment failed.")
```
Comment (yuriolisa) [inline internal/sub_reconcilers/reconcile_deployment_updates.go:132]: Could you please specify better the error?

**Review by yuriolisa — CHANGES_REQUESTED** [review]

File: internal/sub_reconcilers/reconcile_deployment_updates.go (hunk 1 above)
Comment (skattoju) [reply internal/sub_reconcilers/reconcile_deployment_updates.go:127]: mmm there is only one.. but I guess can use a different word than operand

File: internal/sub_reconcilers/reconcile_deployment_updates.go (hunk 2 above)
Comment (skattoju) [reply internal/sub_reconcilers/reconcile_deployment_updates.go:132]: yes this is too vague

**Pull request merged**